### PR TITLE
SPA Environmental Specialist

### PR DESCRIPTION
**Pull request by pheonixstorm**
@HammerGS review the rules section CO p74 and TO:AR table p34, rules p54-60 and make sure I didn't miss anything, @Windchild292  or @NickAragua check the code, there are probably a few places it could be cleaned up but this is a lot of mind numbing amount of variation of the same if statements everywhere...

### CODE REVIEW

File: megamek/i18n/megamek/common/options/messages.properties [hunk 1]
```diff
@@ -571,6 +571,8 @@ PilotOptionsInfo.option.range_master.displayableName= RangeMaster (CamOps)
 PilotOptionsInfo.option.range_master.description=Range modifiers are swapped with short range (Medium, Long, or Extreme).
 PilotOptionsInfo.option.sniper.displayableName= Sniper (CamOps)
 PilotOptionsInfo.option.sniper.description=Range penalties are halved.
+PilotOptionsInfo.option.env_specialist.displayableName=Environmental Specialist (CamOps)
+PilotOptionsInfo.option.env_specialist.description=Environmental specialists are adept in certain planetary conditions.\nThis gives the unit bonuses to PSR and movement penalties (halves MP/PSR) as well tohit rolls (-1).
```
Comment (Windchild292) [inline megamek/i18n/megamek/common/options/messages.properties:575]:
```suggestion
PilotOptionsInfo.option.env_specialist.description=Environmental specialists are adept in certain planetary conditions.\nThis gives the unit bonuses to PSR and movement penalties (halves MP / PSR) as well to hit rolls (-1).
```

File: megamek/src/megamek/common/Aero.java [hunk 2]
```diff
@@ -350,6 +350,10 @@ public int getWalkMP(boolean gravity, boolean ignoreheat, boolean ignoremodulara
             if (weatherMod != 0) {
                 j = Math.max(j + weatherMod, 0);
             }
+            if(getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```
Comment (Windchild292) [inline megamek/src/megamek/common/Aero.java:353]:
```suggestion
            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```

Comment (Windchild292) [reply megamek/src/megamek/common/Aero.java:353]: This is an accessibility issue

File: megamek/src/megamek/common/Aero.java [hunk 3]
```diff
@@ -350,6 +350,10 @@ public int getWalkMP(boolean gravity, boolean ignoreheat, boolean ignoremodulara
...
+                    && game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Aero.java:354]:
```suggestion
                    && (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13)) {
```

Comment (Windchild292) [reply megamek/src/megamek/common/Aero.java:354]: What if it's stronger than that?

File: megamek/src/megamek/common/BattleArmor.java [hunk 4]
```diff
@@ -450,6 +450,16 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
                 j = Math.max(j + weatherMod, 0);
             }
         }
+        if (null != game) {
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_GUSTING_RAIN && getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_RAIN)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/BattleArmor.java:454]:
```suggestion
            if ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_GUSTING_RAIN) && getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_RAIN)) {
```

File: megamek/src/megamek/common/BattleArmor.java [hunk 5]
```diff
@@ -450,6 +450,16 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
...
+                j += 1;
+            }
+            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```
Comment (Windchild292) [inline megamek/src/megamek/common/BattleArmor.java:457]:
```suggestion
            }

            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```

File: megamek/src/megamek/common/BattleArmor.java [hunk 6]
```diff
@@ -450,6 +450,16 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
...
+                    && (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STRONG_GALE
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STORM)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/BattleArmor.java:459]:
```suggestion
                    && ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STRONG_GALE)
                    || (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STORM))) {
```

File: megamek/src/megamek/common/BipedMech.java [hunk 7]
```diff
@@ -172,6 +172,10 @@ public int getWalkMP(boolean gravity, boolean ignoreheat, boolean ignoremodulara
             if (weatherMod != 0) {
                 wmp = Math.max(wmp + weatherMod, 0);
             }
+            if(getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
+                    && game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/BipedMech.java:176]:
```suggestion
            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
                    && (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13)) {
```

Same, what about stronger wind?

File: megamek/src/megamek/common/VTOL.java [hunk 8]
```diff
@@ -575,6 +575,18 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
             if (weatherMod != 0) {
                 j = Math.max(j + weatherMod, 0);
             }
+            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_SNOW)) {
+                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM) {
+                    j += 2;
+                }
```
Comment (Windchild292) [inline megamek/src/megamek/common/VTOL.java:581]: Space before between ifs, and between all equals as per our code style guide.

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 9]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
         }
         
         // VSP Lasers
+
+        //SPA Environmental Specialist
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3029]:
```suggestion
        // SPA Environmental Specialist
```

Accessibility issue

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 10]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+        // Fog Specialist
+        if (ae.getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_FOG)
+                && wtype != null && wtype.hasFlag(WeaponType.F_ENERGY) && !game.getBoard().inSpace()
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3032]:
```suggestion
                && (wtype != null) && wtype.hasFlag(WeaponType.F_ENERGY) && !game.getBoard().inSpace()
```

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 11]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+        // Light Specialist
+        if (ae.getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_LIGHT)) {
+            if (!te.isIlluminated()
+                    && ((game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_DUSK)
+                    || (game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_FULL_MOON)
+                    || (game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_MOONLESS)
+                    || (game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_PITCH_BLACK))) {
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.LightSpec"));
+            } else if (te.isIlluminated()
+                    && game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_PITCH_BLACK) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3046]:
```suggestion
                    && (game.getPlanetaryConditions().getLight() == PlanetaryConditions.L_PITCH_BLACK)) {
```

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 12]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.LightSpec"));
+            }
+        }
+
+        // Rain Specialist
+        if (ae.getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_RAIN)) {
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_LIGHT_RAIN
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3053]:
```suggestion
            if ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_LIGHT_RAIN)
```

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 13]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                    && ae.isConventionalInfantry()) {
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.RainSpec"));
+            }
+            if  (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_MOD_RAIN
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_HEAVY_RAIN
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_GUSTING_RAIN
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_DOWNPOUR) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3060]: Spacing and brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 14]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.RainSpec"));
+            }
+        }
+
+        // Snow Specialist
+        if (ae.getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_SNOW)) {
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_LIGHT_SNOW
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3067]: brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 15]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                    && ae.isConventionalInfantry()) {
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.SnowSpec"));
+            }
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM
+                    && (wtype !=null && wtype.hasFlag(WeaponType.F_MISSILE))) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3072]: Spacing and brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 16]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.SnowSpec"));
+            }
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_SLEET
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_SNOW_FLURRIES
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_MOD_SNOW
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_HEAVY_SNOW) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3078]: Spacing and brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 17]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.SnowSpec"));
+            }
+        }
+
+        // Wind Specialist
+        if (ae.getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)) {
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_MOD_GALE
+                    && (wtype !=null && wtype.hasFlag(WeaponType.F_MISSILE))) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3086]: brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 18]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.SnowSpec"));
+            }
+            if (wtype !=null && wtype.hasFlag(WeaponType.F_MISSILE) && wtype.hasFlag(WeaponType.F_BALLISTIC)
+                    && (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STRONG_GALE
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STORM)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3091]: Spacing and brackets

File: megamek/src/megamek/common/actions/WeaponAttackAction.java [hunk 19]
```diff
@@ -3025,6 +3025,79 @@ private static ToHitData compileWeaponToHitMods(Game game, Entity ae, Entity spo
...
+                toHit.addModifier(-1, Messages.getString("WeaponAttackAction.WindSpec"));
+            }
+            if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13
+                    || game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F4) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/actions/WeaponAttackAction.java:3095]: Spacing and brackets

File: megamek/src/megamek/common/TripodMech.java [hunk 20]
```diff
@@ -218,6 +218,10 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
             if (weatherMod != 0) {
                 wmp = Math.max(wmp + weatherMod, 0);
             }
+            if(getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```
Comment (Windchild292) [inline megamek/src/megamek/common/TripodMech.java:221]:
```suggestion
            if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)
```

File: megamek/src/megamek/common/TripodMech.java [hunk 21]
```diff
@@ -218,6 +218,10 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
...
+                    && game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/TripodMech.java:222]:
```suggestion
                    && (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_TORNADO_F13)) {
```

**Review by Windchild292 — CHANGES_REQUESTED** [review]
Please fix the spacing and bracket issues across the PR before I finish the review

File: megamek/src/megamek/common/Aero.java (hunk 3 above)
Comment (pheonixstorm) [reply megamek/src/megamek/common/Aero.java:354]: F4 which is Mech only, no other unit type allowed.

File: megamek/src/megamek/common/BipedMech.java (hunk 7 above)
Comment (pheonixstorm) [reply megamek/src/megamek/common/BipedMech.java:176]: F4 for mech is walk only

**Comment by pheonixstorm** [comment]
Changes made, code cleaned up, made a few improvements. Time to finish the review so I can put up the PR for MekHQ

File: megamek/src/megamek/common/Entity.java [hunk 22]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
+                    || (conditions.getWeather() == PlanetaryConditions.WE_ICE_STORM))
                     && isAirborneVTOLorWIGE()) {
                 roll.addModifier(-1, Messages.getString("WeaponAttackAction.SnowSpec"));
             }
         }
+
         if (!hasAbility(OptionsConstants.UNOFF_ALLWEATHER)
                 && getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)) {
-            if (conditions.getWeather() == PlanetaryConditions.WI_MOD_GALE && isAirborneVTOLorWIGE()) {
+            if ((conditions.getWeather() == PlanetaryConditions.WI_MOD_GALE && isAirborneVTOLorWIGE())) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7118]:
```suggestion
            if ((conditions.getWeather() == PlanetaryConditions.WI_MOD_GALE) && isAirborneVTOLorWIGE()) {
```

File: megamek/src/megamek/common/Entity.java [hunk 23]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
                 roll.addModifier(-1, Messages.getString("WeaponAttackAction.WindSpec"));
             }
-            if (conditions.getWeather() == PlanetaryConditions.WI_STRONG_GALE) {
-                if (this instanceof Mech || isAirborne()
-                        || isAirborneVTOLorWIGE() || getMovementMode() == EntityMovementMode.HOVER) {
+
+            if ((conditions.getWeather() == PlanetaryConditions.WI_STRONG_GALE)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7122]:
```suggestion
            if (conditions.getWeather() == PlanetaryConditions.WI_STRONG_GALE) {
```

File: megamek/src/megamek/common/Entity.java [hunk 24]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
+            if ((conditions.getWeather() == PlanetaryConditions.WE_DOWNPOUR)
+                    ||(conditions.getWeather() == PlanetaryConditions.WE_HEAVY_RAIN)) {
                 roll.addModifier(-1, Messages.getString("WeaponAttackAction.RainSpec"));
             }
         }
+
         if (!hasAbility(OptionsConstants.UNOFF_ALLWEATHER)
                 && getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_SNOW)) {
-            if (conditions.getWeather() == PlanetaryConditions.WE_HEAVY_SNOW) {
+            if ((conditions.getWeather() == PlanetaryConditions.WE_HEAVY_SNOW)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7105]:
```suggestion
            if (conditions.getWeather() == PlanetaryConditions.WE_HEAVY_SNOW) {
```

File: megamek/src/megamek/common/Entity.java [hunk 25]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
+                if ((this instanceof Mech) || isAirborne()
+                        || isAirborneVTOLorWIGE() || (getMovementMode() == EntityMovementMode.HOVER)) {
                     roll.addModifier(-1, Messages.getString("WeaponAttackAction.WindSpec"));
                 }
             }
-            if (conditions.getWeather() == PlanetaryConditions.WI_STORM) {
-                if (this instanceof Mech || isAirborneVTOLorWIGE()
-                        || getMovementMode() == EntityMovementMode.HOVER) {
+
+            if ((conditions.getWeather() == PlanetaryConditions.WI_STORM)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7129]:
```suggestion
            if (conditions.getWeather() == PlanetaryConditions.WI_STORM) {
```

File: megamek/src/megamek/common/Entity.java [hunk 26]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
                     roll.addModifier(-2, Messages.getString("WeaponAttackAction.WindSpec"));
                 }
+
                 if (isAirborne()) {
                     roll.addModifier(-1, Messages.getString("WeaponAttackAction.WindSpec"));
                 }
             }
-            if (conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F13) {
+
+            if ((conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F13)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7140]:
```suggestion
            if (conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F13) {
```

File: megamek/src/megamek/common/Entity.java [hunk 27]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
                 roll.addModifier(-2, Messages.getString("WeaponAttackAction.WindSpec"));
             }
-            if (conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F4) {
+
+            if ((conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F4)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Entity.java:7144]:
```suggestion
            if (conditions.getWeather() == PlanetaryConditions.WI_TORNADO_F4) {
```

File: megamek/src/megamek/common/Infantry.java [hunk 28]
```diff
@@ -379,30 +379,35 @@ && isConventionalInfantry()) {
...
+                if (((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_SNOW_FLURRIES)
+                        || (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM))
+                        && (getOriginalWalkMP() != 0)) {
                     mp += 1;
                 }
             }
+
             if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_WIND)) {
-                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_MOD_GALE) {
+                if ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_MOD_GALE)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Infantry.java:400]:
```suggestion
                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_MOD_GALE) {
```

File: megamek/src/megamek/common/Entity.java (hunk 24 above)
Comment (Windchild292) [reply megamek/src/megamek/common/Entity.java:7105]: Single brackets are all that are needed for a single statement

File: megamek/src/megamek/common/VTOL.java [hunk 29]
```diff
@@ -575,15 +575,17 @@ public int getWalkMP(boolean gravity, boolean ignoreheat,
             if (weatherMod != 0) {
                 j = Math.max(j + weatherMod, 0);
             }
+
             if (getCrew().getOptions().stringOption(OptionsConstants.MISC_ENV_SPECIALIST).equals(Crew.ENVSPC_SNOW)) {
-                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM) {
+                if ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM)) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/VTOL.java:580]:
```suggestion
                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WE_ICE_STORM) {
```

File: megamek/src/megamek/common/Infantry.java [hunk 30]
```diff
@@ -379,30 +379,35 @@ && isConventionalInfantry()) {
...
                     mp += 1;
                 }
-                if (game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STRONG_GALE
-                        && (mp !=0 || getMovementMode().isMotorizedInfantry())) {
+
+                if ((game.getPlanetaryConditions().getWeather() == PlanetaryConditions.WI_STRONG_GALE)
+                        && ((mp !=0) || getMovementMode().isMotorizedInfantry())) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Infantry.java:405]:
```suggestion
                        && ((mp != 0) || getMovementMode().isMotorizedInfantry())) {
```

**Review by Windchild292 — CHANGES_REQUESTED** [review]
Just a few more cleanups, the main ones being overzealous bracket implementations

File: megamek/src/megamek/common/Entity.java [hunk 31]
```diff
@@ -7085,56 +7085,63 @@ public PilotingRollData addConditionBonuses(PilotingRollData roll,
...
-                if (this instanceof Mech || isAirborne()
-                        || getMovementMode() == EntityMovementMode.WHEELED
-                        || getMovementMode() == EntityMovementMode.TRACKED) {
+                if ((this instanceof Mech) || isAirborne()
+                        || (getMovementMode() == EntityMovementMode.WHEELED)
+                        || (getMovementMode() == EntityMovementMode.TRACKED)) {
                     roll.addModifier(-1, Messages.getString("WeaponAttackAction.RainSpec"));
                 }
-                if (isAirborneVTOLorWIGE() || getMovementMode() == EntityMovementMode.HOVER) {
+                if (isAirborneVTOLorWIGE() || (getMovementMode() == EntityMovementMode.HOVER)) {
```
[the remark this answers is not in the document]
Comment (pheonixstorm) [reply megamek/src/megamek/common/Entity.java:7093]: Hover and other ground vehicles don't have an is????

Would make for a good cleanup after this is done. Not sure where else getMovementMode() is used.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**